lib/site-builder.js

```diff
@@ -70,18 +71,31 @@ SiteBuilder.prototype.spawn = function(path, args) {
 };
 
 SiteBuilder.prototype.build = function(done) {
-  var that = this;
-  var doBuild = function(lockedOperationDone) {
-    fs.exists(that.opts.sitePath, function(exists) {
-      (exists ? that.syncRepo() : that.cloneRepo())
-        .then(function() { return that.checkForFile('_config.yml'); })
-        .then(function(useJekyll) {
-          return (useJekyll ? that._buildJekyll() : that._rsync());
-        })
-        .then(lockedOperationDone, lockedOperationDone);
+  var builder = this,
+      doBuild,
+      finish;
+
+  doBuild = function() {
+    return new Promise(function(resolve) {
+      fs.exists(builder.opts.sitePath, function(exists) {
+        resolve(exists ? builder.syncRepo() : builder.cloneRepo());
+      });
+    })
+    .then(function() { return builder.checkForFile('_config.yml'); })
+    .then(function(useJekyll) {
+      return (useJekyll ? builder._buildJekyll() : builder._rsync());
     });
   };
-  this.updateLock.doLockedOperation(doBuild, this.generateBuildDone(done));
+
+  finish = function(err) {
+    return builder.finishBuild()
+      .then(function() {
+        done(err);
```

Comment:
Not exactly, but I've updated both this function and `finishBuild()` in a way that is hopefully more straightforward and readable.